### ISSUE TITLE
No longer working with the latest beta (v2.2.4)

### ISSUE THREAD

**Issue by harshcurious** [benji300/joplin-note-tabs#49]
The plugin no longer works with the latest Beta([v2.2.4](https://github.com/laurent22/joplin/releases/tag/v2.2.4))!
Here is the error from the console:
```
15:13:12: joplin.plugins: Uncaught exception in plugin "joplin.plugin.note.tabs": Error: "joplin.plugin.note.tabs": No longer supported: joplin.settings.registerSetting() is deprecated in favour of joplin.settings.registerSettings() (deprecated since version 1.8)
    at Plugin.deprecationNotice (/usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/Plugin.js:117)
    at JoplinSettings.<anonymous> (/usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/api/JoplinSettings.js:85)
    at Generator.next (<anonymous>)
    at /usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/api/JoplinSettings.js:8
    at new Promise (<anonymous>)
    at __awaiter (/usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/api/JoplinSettings.js:4)
    at JoplinSettings.registerSetting (/usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/api/JoplinSettings.js:84)
    at Object.<anonymous> (/usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/utils/executeSandboxCall.js:50)
    at Generator.next (<anonymous>)
    at /usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/services/plugins/utils/executeSandboxCall.js:8
log @ /usr/share/joplin/resources/app.asar/node_modules/@joplin/lib/Logger.js:188
```
Here is a discussion that should help with the error: [https://discourse.joplinapp.org/t/19278](https://discourse.joplinapp.org/t/19278)

**Comment by harshcurious** [comment]
I have been looking at the code and you just need to convert [Line 190](https://github.com/benji300/joplin-note-tabs/blob/03a7b799071542891362eb4822f26fc4bb479e5b/src/settings.ts#L190) onward (consisting of several `register.Setting` commands) into the new `register.Settings` format.

[From the API reference](https://joplinapp.org/api/references/plugin_api/classes/joplinsettings.html):

### registerSettings

-   registerSettings(settings: Record<string, [SettingItem](https://joplinapp.org/api/references/plugin_api/classes/joplinsettings.html../interfaces/settingitem.html)\>): Promise<void\>

I am not very familiar with TypeScript, otherwise I would have tried correcting it myself 😔

**Comment by harshcurious** [comment]
[Here](https://github.com/laurent22/joplin/blob/e9d590169b7018fef0dd4b08a9bfd1fdfbf51ebf/packages/app-cli/tests/support/plugins/settings/src/index.ts#L11-L52) is an example of the new implementation:
```ts
		await joplin.settings.registerSettings({
			'myCustomSetting': {
				value: 123,
				type: SettingItemType.Int,
				section: 'myCustomSection',
				public: true,
				label: 'My Custom Setting',
			},


			'multiOptionTest': {
				value: 'en',
				type: SettingItemType.String,
				section: 'myCustomSection',
				isEnum: true,
				public: true,
				label: 'Multi-options test',
				options: {
					'en': 'English',
					'fr': 'French',
					'es': 'Spanish',
				},
			},


			'mySecureSetting': {
				value: 'hunter2',
				type: SettingItemType.String,
				section: 'myCustomSection',
				public: true,
				secure: true,
				label: 'My Secure Setting',
			},


			'myFileSetting': {
				value: 'abcd',
				type: SettingItemType.String,
				section: 'myCustomSection',
				public: true,
				label: 'My file setting',
				description: 'This setting will be saved to settings.json',
				['storage' as any]: 2, // Should be `storage: SettingStorage.File`
			},
		});
```

**Comment by benji300** [comment]
Duplicate of #48.

**Comment by benji300** [comment]
Thanks for reporting. I've already an implementation on a branch.

**Comment by harshcurious** [comment]
> Duplicate of #48.

Sorry about that, it came up after I had opened this one!

> Thanks for reporting. I've already an implementation on a branch.

Yay! Thanks a lot for the awesome plugin and for continuously maintaining it!💜

**Comment by benji300** [comment]
No worries @harshcurious... I just closed this one because it got the "higher" issue number 😉 
Thanks for your feedback 😄 

FYI: I've published a new version of the plugin ([v1.4.0](https://github.com/benji300/joplin-note-tabs/releases/tag/v1.4.0)) which contains the changes and is ready for Joplin v2.2.4 and newer.
